tests/frameworks/huggingface/serving.py

```diff
@@ -0,0 +1,145 @@
+# Copyright 2019 Iguazio
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+from typing import Dict
+
+import numpy as np
+import pytest
+
+import mlrun
+
+CLASS_NAME = "mlrun.frameworks.huggingface.HuggingFaceModelServer"
+
+# The PIPELINES list below contains all the NLP tasks that exists in the huggingface framework.
+# Each example is a dictionary that include:
+# - task        = the name of task
+# - example     = input for prediction
+# - result_keys = the keys that need to appear in the prediction.
+
+PIPELINES = [
+    {
+        "task": "sentiment-analysis",
+        "example": "We are very happy to show you the 🤗 Transformers library.",
+        "result_keys": ["label", "score"],
+    },
+    {
+        "task": "text-generation",
+        "example": {
+            "text_inputs": "Hello, I'm a language model",
+            "max_length": 20,
+            "num_return_sequences": 1,
+        },
+        "result_keys": ["generated_text"],
+    },
+    {
+        "task": "ner",
+        "example": "My name is Wolfgang",
+        "result_keys": ["entity", "score", "index", "word", "start", "end"],
+    },
+    {
+        "task": "question-answering",
+        "example": {
+            "question": "Where do I live?",
+            "context": "My name is Merve and I live in İstanbul.",
+        },
+        "result_keys": ["score", "start", "end", "answer"],
+    },
+    {
+        "task": "fill-mask",
+        "example": "Paris is the <mask> of France.",
+        "result_keys": ["score", "token", "token_str", "sequence"],
+    },
+    {
+        "task": "summarization",
+        "example": "Paris is the capital and most populous city of France,"
+        " with an estimated population of 2,175,601 residents as of 2018,"
+        " in an area of more than 105 square kilometres (41 square miles)."
+        " The City of Paris is the centre and seat of government of the region"
+        " and province of Île-de-France, or Paris Region, which has an estimated population of 12,174,880,"
+        " or about 18 percent of the population of France as of 2017.",
+        "result_keys": ["summary_text"],
+    },
+    {
+        "task": "translation_en_to_fr",
+        "example": "How old are you?",
+        "result_keys": ["translation_text"],
+    },
+]
```

Comment:
Looks like a copy-paste from HuggingFace, I'm not sure it is a good thing. Please consult with Yaron if he approves.